python/tank/util/metrics.py

```diff
@@ -324,100 +327,196 @@ def _dispatch(self, metrics):
-        self._data = data
+    TODO: How about event type which is defined in the 'Shotgun Event Taxonomy' document as event property???
+    """
 
-    def __str__(self):
-        """Readable str representation of the metric."""
-        return "%s: %s" % (self.__class__, self._data)
+    """
+    Use groups to distinguish logical parts of your application.
+    Event groups help with prioritizing key components of your application
+    to instrument and can assist with event naming.
+    """
 
-    @property
-    def data(self):
-        """The underlying data this metric represents."""
-        return self._data
+    GROUP_TRIAL_SIGNUP = "Trial Signup"
+    GROUP_USER_SIGNUP = "User Signup"
+    GROUP_APP = "App"
+    GROUP_ONBOARDING = "Onboarding"
+    GROUP_ACCOUNT_SETTINGS = "Account Settings"
+    GROUP_PROJECTS = "Projects"
+    GROUP_ENTITIES = "Entities"
+    GROUP_TASKS = "Tasks"
+    GROUP_PEOPLE = "People"
+    GROUP_MEDIA = "Merdia"
+    GROUP_NOTES = "Notes"
+    GROUP_BILLING = "Billing"
+    GROUP_TOOLKIT = "Toolkit"
+
+    # Event property keys
+    KEY_ACTION_TITLE = "Action Title"
+    KEY_APP = "App"
+    KEY_APP_VERSION = "App Version"
+    KEY_COMMAND = "Command"
+    KEY_ENGINE = "Engine"
+    KEY_ENGINE_VERSION = "Engine Version"
+    KEY_ENTITY_TYPE = "Entity Type"
+    KEY_HOST_APP = "Host App"
+    KEY_HOST_APP_VERSION = "Host App Version"
+    KEY_PUBLISH_TYPE = "Publish Type"
+
+    def __init__(self, group, name, properties=None):
+        """
+        Initialize a metric event using the specified parameters. Use this helper class
+        to create a suitable metric structure to be used with the
+
+        :param str group: A group or category this metric event falls into.
+        Although any values can be used, we encourage usage of the GROUP_*
+        definitions above.
+=
+        :param: str name: A short descriptive event name or performed action.
+            The complete list can be found in the 'Shotgun Event Taxonomy' document.
+            Below are a few examples:
+                'Viewed Login Page'
+                'Logged In'
+                'Created Project'
+                'Toggled Project Favorite'
+                'Edited Task Status'
+                'Read Inbox Item'
+
+        :param: dict properties: An optional dictionary of extra properties to be attached to the metric event.
+        """
 
+        """
+        We also add an empty event property dictionary that will be populated
+        with either or both specified properties or add system info properties.
+        """
+        self._data = {
+            "event_group": str(group),
+            "event_name": str(name),
+            "event_property": {}
+        }
 
-class UserActivityMetric(ToolkitMetric):
-    """Convenience class for a user activity metric."""
+        if properties:
+            if not type(properties) is dict:
+                raise TypeError("The `properties` parameter must be None or a dictionary")
```

Comment:
even with the EventMetric.log class method, shouldn't we enforce that the 3rd param be a dictionary ?